pandas/_libs/groupby.pyx

```diff
@@ -604,7 +625,23 @@ def group_sum(
             for i in range(ncounts):
                 for j in range(K):
                     if nobs[i, j] < min_count:
-                        out[i, j] = NAN
+                        # if we are integer dtype, not is_datetimelike, and
+                        #  not uses_mask, then getting here implies that
+                        #  counts[i] < min_count, which means we will
```

Comment:
Is it worth (or feasible) to have the actual check in the code?

Reply:
Not really, we can’t set nan into int dtypes, so we have to use a placeholder anyway